Assets/Scripts/GraphService.cs

```diff
@@ -12,6 +11,19 @@ public class GraphService : MonoBehaviour
     public EdgeGameObjectFactory edgeGameObjectFactory;
     public NodeGameObjectFactory nodeGameObjectFactory;
     public Graph Graph { get; private set; }
+    private bool _labelVisibility = false;
+    public bool LabelVisibility
```

Comment:
I don't like this approach - LabelVisibility is not a property of GraphService, rather a setting set by graphservice. Cases like that should stay as methods.
Also as a rule of thumb, comlicated logic should be avoided in properties, here iterating over each node and edge might take a lot of time. A good reason to keep as method

Reply:
you are probably right about complex logic in properties but i thougth that knowlage if labels are visible is necesarry, 
set method was already there, so I added also get
before that new nodes were added always with label visible even that labels were turned off